### PR TITLE
'sk test' command upgrades

### PR DESCRIPTION
**Pull request by ShaneBeee**
### Description
This PR aims to enhance the `/sk test` command:
- Add test script names to tab complete (from the Test script directory):
Screenshot 2024-12-23 at 1 56 06 PM

- Allow using `/sk test all` to run all of your test scripts:
Screenshot 2024-12-23 at 1 56 52 PM

- Allow running tests for a directory:
Screenshot 2024-12-23 at 2 08 39 PM

- Log results to file (since the result path is required, let's use it):
Screenshot 2024-12-23 at 2 46 30 PM

- Use pretty printing for json output (easier to read than one super long line):
(This is both for the test command as well as regular test runs)
BEFORE:
Screenshot 2024-12-23 at 3 21 48 PM

AFTER:
Screenshot 2024-12-23 at 3 19 45 PM

- Fix a unicode character issue in the outputs:
BEFORE:
Screenshot 2024-12-23 at 3 24 37 PM

AFTER:
Screenshot 2024-12-23 at 3 24 58 PM

- Added some color to the command output (thanks to Sovde's suggestion):
Screenshot 2024-12-26 at 3 16 36 PM

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** none

### CODE REVIEW

File: src/main/java/ch/njol/skript/SkriptCommand.java [hunk 1]
```diff
@@ -406,10 +408,21 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
...
-						Arrays.stream(args).skip(1).collect(Collectors.joining(" ")) + ".sk"
-					).toFile();
-					TestMode.lastTestFile = scriptFile;
+					if (args[1].equalsIgnoreCase("all")) {
+						scriptFile = TestMode.TEST_DIR.toFile();
+					} else {
+						String fileName = Arrays.stream(args).skip(1).collect(Collectors.joining(" "));
+
+						// Add .sk if the file is not a directory and doesn't currently have .sk
+						if (!fileName.endsWith(".sk") && !fileName.endsWith("/"))
```
[the remark this answers is not in the document]
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/SkriptCommand.java:417]: At the point of this check, its just a file name, so I cant check if its a directory.

This is going off of what is provided from tab complete:
Screenshot 2024-12-24 at 10 10 02 AM
At this point, scripts end with `.sk` whereas directories end with `/`

[the remark this answers is not in the document]
Comment (sovdeeth) [reply src/main/java/ch/njol/skript/SkriptCommand.java:417]: I don't think that'd match reload's behavior though. reload's smart enough to detect directories that don't end in /, so I think this command should too.

**Review by sovdeeth — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/SkriptCommand.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/SkriptCommand.java:417]: I'll take a peak at the reload code later and see if I can replicate.

[the remark this answers is not in the document]
Comment (ShaneBeee) [reply src/main/java/ch/njol/skript/SkriptCommand.java]: Ok, I saw that "reload" uses some static methods.
I just had to refactor the methods (and add some overloads) to just grab the scripts from the test directory using those methods.
This all works splendidly now.

**Review by sovdeeth — APPROVED** [review]

**Review by cheeezburga — APPROVED** [review]

File: src/main/java/ch/njol/skript/SkriptCommand.java [hunk 2]
```diff
@@ -464,8 +481,13 @@ public boolean onCommand(CommandSender sender, Command command, String label, St
 
 	@Nullable
 	private static File getScriptFromArgs(CommandSender sender, String[] args) {
+		return getScriptFromArgs(sender, args, Skript.getInstance().getScriptsFolder());
+	}
+
+	@Nullable
+	private static File getScriptFromArgs(CommandSender sender, String[] args, File directoryFile) {
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/SkriptCommand.java:488]:
```suggestion
	private static @Nullable File getScriptFromArgs(CommandSender sender, String[] args, File directoryFile) {
```

File: src/main/java/ch/njol/skript/SkriptCommand.java [hunk 3]
```diff
@@ -477,6 +499,11 @@ private static File getScriptFromArgs(CommandSender sender, String[] args) {
 
 	@Nullable
 	public static File getScriptFromName(String script) {
+		return getScriptFromName(script, Skript.getInstance().getScriptsFolder());
+	}
+
+	@Nullable
+	public static File getScriptFromName(String script, File directoryFile) {
```
Comment (Efnilite) [inline src/main/java/ch/njol/skript/SkriptCommand.java:506]:
```suggestion
	public static @Nullable File getScriptFromName(String script, File directoryFile) {
```

**Review by Efnilite — APPROVED** [review]

**Pull request merged**